src/hooks/NotificationsProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import { createContext, FC, ReactNode, useState } from "react"
+import { UserNotification } from "../types/lernfair/Notification"
+
+type Notification = Partial<UserNotification>
+
+type NotificationContextValue = {notifications: Notification[], dispatchNewNotificationIds: Function};
+export const NotificationsContext = createContext<NotificationContextValue>({notifications: [], dispatchNewNotificationIds: () => null})
+
+type State = { notifications: Notification[] }
+
+export const NotificationsProvider: FC<{ children: ReactNode }> =
+  ({ children })=> {
+    const [{notifications}, setState] = useState<State>({notifications: []})
```

Comment:
What about: 
```suggestion
    const [notifications, setState] = useState<State>([])
```

The state would then be `Notification[]`